classes/module/Module.php

```diff
@@ -3436,26 +3436,22 @@ public function isSymfonyContext()
     }
 
     /**
-     * Access the Symfony Container if we are in Symfony Context.
-     * Note: in this case, we must get a container from SymfonyContainer class.
-     * Note: if not in Symfony context, fallback to legacy Container for FO/BO.
+     * Access a service from the container (found from the controller or the context
+     * depending on cases). It uses ContainerFinder to find the appropriate container.
      *
      * @param string $serviceName
      *
      * @return object|false if a container is not available, it returns false
      */
     public function get($serviceName)
     {
-        if ($this->isSymfonyContext()) {
-            if (null === $this->container) {
-                $this->container = SymfonyContainer::getInstance();
-            }
-
-            return $this->container->get($serviceName);
+        if (null === $this->container) {
+            $finder = new ContainerFinder($this->context);
+            $this->container = $finder->getContainer();
         }
 
-        if ($this->context->controller instanceof Controller) {
-            return $this->context->controller->get($serviceName);
+        if (null !== $this->container) {
```

Comment:
`$this->container` cannot be null, a `ContainerNotFoundException` is raised if there is no container available.

Reply:
It shouldn't be null, but sometime it might be I guess, if the context is badly initialized
Although I would love to raise this exception, we can't do that because it would probably cause some BC breaks and unexpected behaviour